.github/workflows/nightly.yml

```diff
@@ -0,0 +1,79 @@
+name: nightly
+
+on:
+  schedule:
+    - cron: '0 0 * * *'  # every day at midnight
+
+env:
+  CARGO_TERM_COLOR: always
+  # Disable incremental compilation.
+  #
+  # Incremental compilation is useful as part of an edit-build-test-edit cycle,
+  # as it lets the compiler avoid recompiling code that hasn't changed. However,
+  # on CI, we're not making small edits; we're almost always building the entire
+  # project from scratch. Thus, incremental compilation on CI actually
+  # introduces *additional* overhead to support making future builds
+  # faster...but no future builds will ever occur in any given CI environment.
+  #
+  # See https://matklad.github.io/2021/09/04/fast-rust-builds.html#ci-workflow
+  # for details.
+  CARGO_INCREMENTAL: 0
+  # Allow more retries for network requests in cargo (downloading crates) and
+  # rustup (installing toolchains). This should help to reduce flaky CI failures
+  # from transient network timeouts or other issues.
+  CARGO_NET_RETRY: 10
+  RUSTUP_MAX_RETRIES: 10
+  # Don't emit giant backtraces in the CI logs.
+  RUST_BACKTRACE: short
+
+jobs:
+  cargo-udeps:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install minimal nightly
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+      - name: Install cargo-udeps, and cache the binary
+        uses: baptiste0928/cargo-install@v1
+        with:
+          crate: cargo-udeps
+      - name: run cargo-udeps
+        run: cargo +nightly udeps
```

Comment:
Early feedback to developers on whether they added a needless dependency is extremely valuable. We have many PRs where we include such a needless crate and end up not using it. When it's 24h later and merged on main is not a good time to herd cats and assemble the devs of the prior day to clean up added dependencies they didn't really need.

Reply:
Actually, if we can run that on stable, I'm a happy camper!